cmd/bucket-handlers.go

```diff
@@ -549,10 +550,15 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 	for _, dobj := range deletedObjects {
 		if replicateDeletes {
 			if dobj.DeleteMarkerReplicationStatus == string(replication.Pending) || dobj.VersionPurgeStatus == Pending {
-				globalReplicationState.queueReplicaDeleteTask(DeletedObjectVersionInfo{
+				dv := DeletedObjectVersionInfo{
 					DeletedObject: dobj,
 					Bucket:        bucket,
-				})
+				}
+				if replicateSync {
+					replicateDelete(ctx, dv, objectAPI)
```

Comment:
we should also return error to the client if the synchronous replication fails @poornas ?- this way application knows that replication has failed?